### PR TITLE
Create MarkdownService, EmojisService, CodesOfConductService and MetaService 

### LINKED ISSUE #2936: Proposal: Move methods off of Client
Tooling like update-urls and [metadata](https://github.com/google/go-github/issues/2908) could be simpler if API calls were made from methods on a `*Service` receiver. I think it would also make more sense for humans using go-github.

Currently there are handful of Client methods such as that would logically belong on different receivers. I propose we create the appropriate Service types for these.

The changes would be would be:
- Client.Markdown -> MarkdownService.Markdown (maybe rename it to RenderMarkdown while we are at it)
- Client.ListEmojis -> EmojisService.ListEmojis
- Client.ListCodesOfConduct -> CodesOfConductService.ListCodesOfConduct
- Client.GetCodeOfConduct -> CodesOfConductService.GetCodeOfConduct
- Client.APIMeta -> MetaService.APIMeta
- Client.Octocat -> MetaService.Octocat
- Client.Zen -> MetaService.Zen

The original methods would be updated to call their new equivalent and marked deprecated.

### PR DESCRIPTION
**Pull request by WillAbides**
closes #2936

This moves all Client methods from misc.go to new services.

In every case but one, the original Client method was left in place and refactored to call the new method. The exception is `Client.Markdown` because the old method name is now the name of the service field.

### CODE REVIEW

**Review by gmlewis — COMMENTED** [review]
In keeping with our `## Other notes on code organization ##` blurb in CONTRIBUTING.md,
what would you think about moving these new service objects to their own "*.go" files?

File: github/codesofconduct.go
Comment (gmlewis) [inline github/codesofconduct.go:31]: Another thought for you to consider...

Since we are breaking the API anyway with this change (because of `Markdown`), what do you think about reducing the stutter here by renaming `CodesOfConductService.ListCodesOfConduct` to `CodesOfConductService.List`, and same for `Get`, etc?

I looked through this repo to see if there is precedent for this, and there does appear to be:

```bash
$ rg ' List\('
github/codespaces.go
128:func (s *CodespacesService) List(ctx context.Context, opts *ListCodespacesOptions) (*ListCodespaces, *Response, error) {

github/licenses.go
64:func (s *LicensesService) List(ctx context.Context) ([]*License, *Response, error) {

github/gists.go
101:func (s *GistsService) List(ctx context.Context, user string, opts *GistListOptions) ([]*Gist, *Response, error) {

github/gitignore.go
32:func (s *GitignoresService) List(ctx context.Context) ([]string, *Response, error) {

github/pulls.go
146:func (s *PullRequestsService) List(ctx context.Context, owner string, repo string, opts *PullRequestListOptions) ([]*PullRequest, *Response, error) {

github/repos.go
269:func (s *RepositoriesService) List(ctx context.Context, user string, opts *RepositoryListOptions) ([]*Repository, *Response, error) {

github/orgs.go
176:func (s *OrganizationsService) List(ctx context.Context, user string, opts *ListOptions) ([]*Organization, *Response, error) {

github/issues.go
137:func (s *IssuesService) List(ctx context.Context, all bool, opts *IssueListOptions) ([]*Issue, *Response, error) {
```

File: github/markdown.go
Comment (gmlewis) [inline github/markdown.go:44]: Now that this has its own service, would `Render` be a more appropriate verb for the name of the function?
Although the endpoint itself is `markdown` so feel free to ignore this idea.

File: github/meta.go
Comment (gmlewis) [inline github/meta.go:124]: While we are here...

```suggestion
// See also: http://warpspire.com/posts/taste/
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @WillAbides !
I have some questions and a couple tweak requests, please.

**Comment by WillAbides** [comment]
@gmlewis I did the renaming you suggested and took it one further renaming `EmojisService.ListEmojis` to `EmojisService.List`.

I think you were right about renaming Markdown to Render. Even though the endpoint ends with /markdown` the description is "Render a Markdown document".

**Review by gmlewis — APPROVED** [review]
Thank you, @WillAbides !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by WillAbides** [comment]
One more possibility while we are breaking things. Would it make sense to rename `MetaService.APIMeta` to `MetaService.Get`? I would think yes, but my hesitation is that meta is kind of a weird service in that it has functions that don't have anything to do with the metadata.

**Comment by gmlewis** [comment]
> One more possibility while we are breaking things. Would it make sense to rename `MetaService.APIMeta` to `MetaService.Get`? I would think yes, but my hesitation is that meta is kind of a weird service in that it has functions that don't have anything to do with the metadata.

I like it.  Yes, please.

File: github/meta_test.go
Comment (gmlewis) [inline github/meta_test.go:50]:
```suggestion
func TestMetaService_Get(t *testing.T) {
```

File: github/meta_test.go
Comment (gmlewis) [inline github/meta_test.go:60]:
```suggestion
	meta, _, err := client.MetaService.Get(ctx)
```

File: github/meta_test.go
Comment (gmlewis) [inline github/meta_test.go:83]:
```suggestion
		got, resp, err := client.MetaService.Get(ctx)
```

File: github/meta_test.go
Comment (gmlewis) [inline github/meta_test.go:106]:
```suggestion
	got, _, err := client.MetaService.Octocat(ctx, input)
```

File: github/meta_test.go
Comment (gmlewis) [inline github/meta_test.go:117]:
```suggestion
		got, resp, err := client.MetaService.Octocat(ctx, input)
```

File: github/meta_test.go
Comment (gmlewis) [inline github/meta_test.go:138]:
```suggestion
	got, _, err := client.MetaService.Zen(ctx)
```

File: github/meta_test.go
Comment (gmlewis) [inline github/meta_test.go:149]:
```suggestion
		got, resp, err := client.MetaService.Zen(ctx)
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
I think these changes will make it easier to remove the deprecated functions in the future.
Thoughts?

**Comment by abhijit-hota** [comment]
Hey @gmlewis, I see that the PR is waiting on a few test function renames. 

Can I help in anyway to speed things up? I could raise a PR against `WillAbide:nomisc` with the required changes. @WillAbides


P.S., can [this comment](https://github.com/google/go-github/pull/2937#discussion_r1333079385) be resolved? I gather that the [required changes are done](https://github.com/google/go-github/pull/2937#issuecomment-1729718111).

**Comment by gmlewis** [comment]
> P.S., can [this comment](https://github.com/google/go-github/pull/2937#discussion_r1333079385) be resolved? I gather that the [required changes are done](https://github.com/google/go-github/pull/2937#issuecomment-1729718111).

Thanks, @abhijit-hota - marked comment as resolved.

For the remaining items, I was interested in hearing what @WillAbides thinks about the ideas I toseed out, so let's please wait to hear.

**Comment by WillAbides** [comment]
Sorry for the late response. I lost track of this one somehow. @abhijit-hota I appreciate your willingness to step in.

@gmlewis Your suggestions make sense, so I added them. That reduced the code coverage a bit because the deprecated methods are no longer tested. IMO, there isn't a point in testing these anyway.

**Review by gmlewis — APPROVED** [review]
Thank you, @WillAbides !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by abhijit-hota — APPROVED** [review]

**Comment by abhijit-hota** [comment]
LGTM. =)

**Comment by gmlewis** [comment]
Thank you, @abhijit-hota !
Merging.

**Pull request merged**
